testing/kuttl/e2e/delete/21-errors.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: postgres-operator.crunchydata.com/v1beta1
+kind: PostgresCluster
+metadata:
+  name: delete-not-running
+status:
+  instances:
+    - name: instance1
+      readyReplicas: 1
```

Comment:
:thinking: Should this error check happen before the delete step? Is the purpose here to validate the cluster never reached a running state, or is it checking something else?

Reply:
It's to validate that it never runs, yeah. 

(That said, I put this in to make sure that the KUTTL test matches the functionality of the test I'm removing, but I'm still confused -- as I noted on our spreadsheet -- what this test case is really supposed to be testing. Like: maybe there was a possibility that an unrecoverable cluster would retain its finalizer and so never be deleted? I mean, I guess that could happen. I just am not 100% sure.)

Reply:
I suppose part of what I was wondering was if `21-errors.yaml` should effectively move up in the ordering, perhaps to `20-errors.yaml`, so that error check is run before the `delete` step. Otherwise, I think this test makes sense as it covers what was in the original EnvTest (at least implicitly).